bundles/framework/timeseries/TimeseriesAnimationPlugin.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * @class Oskari.mapframework.bundle.timeseries.TimeseriesAnimationPlugin
+ * Handles timeseries enabled layer animation
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.timeseries.TimeseriesAnimationPlugin',
+    function (mapModule) {
+        this._mapModule = mapModule;
+        this._lastFrameLoadTime = 0;
+        this._scheduleNextTimestep = false;
+        this._currentTime = null;
+        this._stepInterval = 60000; // 1 minute
+        this._imageBuffer = Oskari.clazz.create('Oskari.mapframework.domain.ImageBuffer');
+        this._cancelBuffering = null;
+        this._frameInterval = 1000;
+        this._currentLayerId = null;
+    }, {
+        __name : 'TimeseriesAnimationPlugin',
+        _clazz : 'Oskari.mapframework.bundle.timeseries.TimeseriesAnimationPlugin',
+        getCurrentLayerId: function() {
+            return this._currentLayerId;
+        },
+        setMapModule: function (mapModule) {
+            this.mapModule = mapModule;
+        },
+        /**
+         * @method advancePlayback
+         * Schedule new timeseries frame if needed
+         */
+        advancePlayback: function () {
+            var me = this;
+            if (me._scheduleNextTimestep) {
+                var nextStep = me._getNextTimestep(this._stepInterval);
+                if (nextStep) {
+                    var imagesToLoad = me._getTileUrls(nextStep);
+                    var millisToTarget = this._frameInterval - Date.now() + me._lastFrameLoadTime;
+                    var millisTimeout = 5000;
+                    me._cancelBuffering = me._imageBuffer.loadImages(imagesToLoad, millisToTarget, Math.max(millisToTarget, millisTimeout), function (success) {
+                        me._setLayerTimestep(nextStep, true);
+                    });
+                } else {
+                    me._stopTimeseriesPlayback();
+                }
+            }
+            me._scheduleNextTimestep = false;
+        },
+        /**
+         * @method _getTileUrls
+         * Get tile urls for given time
+         * @param {String} time time as ISO-string 
+         */
+        _getTileUrls: function (time) {
```

Comment:
There was no way to get the current tile urls from OpenLayers' public API, so had to resort to this somewhat hacky solution.